src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -147,6 +148,17 @@ public void Submit()
 
         public void Cancel()
         {
+            // Cancel for RenameSession will throw if we try to cancel multiple times. 
+            // We could make sure this doesn't get called multiple times, but that gets difficult
+            // in managing state between why the cancel happened (Press Esc resulting in focus lost
+            // can trigger twice, for example). Rather than relying on correctly handling all the event handlers
+            // from the UI and TextView, just make sure we always only cancel once. 
+            if (_canceled)
+            {
+                return;
+            }
+
+            _canceled = true;
```

Comment:
i would be ok with this being pushed into InlineRenameSessino.  But i don't feel strongly.

Reply:
I like that idea. I initially thought it had a good reason, but tbh even the callers within the session are always having to check the dismissed state. I'll move there